include/ensmallen_bits/parallel_sgd/parallel_sgd_impl.hpp

```diff
@@ -103,29 +141,34 @@ double ParallelSGD<DecayPolicyType>::Optimize(
       {
         // Each instance affects only some components of the decision variable.
         // So the gradient is sparse.
-        arma::sp_mat gradient;
+        BaseGradType gradient;
 
         // Evaluate the sparse gradient.
+        // TODO: support for batch size > 1 could be really useful.
         function.Gradient(iterate, visitationOrder[j], gradient, 1);
 
         // Update the decision variable with non-zero components of the
         // gradient.
         for (size_t i = 0; i < gradient.n_cols; ++i)
         {
           // Iterate over the non-zero elements.
-          for (arma::sp_mat::iterator cur = gradient.begin_col(i);
+          for (typename GradType::iterator cur = gradient.begin_col(i);
```

Comment:
I suggest to declare `cur_end = gradient.end_col(i);` outside of the loop, and then use `cur_end` inside the loop declaration.

Reply:
Agreed, thanks. :+1: